iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -35,7 +35,7 @@ struct iox_ws_storage_t_
 {
     // the value of the array size is the result of the following formula:
     // sizeof(WaitSet) / 8
-    uint64_t do_not_touch_me[2965];
+    uint64_t do_not_touch_me[5958];
```

Comment:
@elfenpiff Here the new values for the types in C-Binding

Reply:
Ouh no, this is evil ... I think we may need some macro magic.

Maybe we need a different approach here. One can also define via cmake macros which are then known in all c/cpp files and I think we require this approach here. In `iceoryx_posh_types.hpp` the variable is then set to this macro value and the size of the waitset/listener are then calculated based on that macro value.

If we do not do this the c binding may crash when the user increases the value. If we introduce a max number we gain nothing in the c binding since here we will use then always the worst case number. I think the macro value approach is the only approach which solves those problems - what do you think?